@here/harp-mapview/lib/PickHandler.ts

```diff
@@ -250,28 +250,39 @@ export class PickHandler {
         intersect: THREE.Intersection,
         pickResult: PickResult
     ) {
+        if (featureData.objInfos === undefined) {
+            return;
+        }
+
         if (pickResult.intersection!.object instanceof MapViewPoints) {
-            pickResult.userData = featureData.objInfos![intersect.index!];
+            pickResult.userData = featureData.objInfos[intersect.index!];
             return;
-        } else if (
-            featureData.objInfos === undefined ||
+        }
+
+        if (
             featureData.starts === undefined ||
-            intersect.faceIndex === undefined
+            featureData.starts.length === 0 ||
+            (intersect.faceIndex === undefined && intersect.index === undefined)
         ) {
             return;
         }
 
-        if (featureData.starts.length > 1) {
-            let objInfosIndex = 0;
-            for (const polygonStartFace of featureData.starts) {
-                if (polygonStartFace > intersect.faceIndex * 3) {
-                    break;
-                }
-                objInfosIndex++;
-            }
-            pickResult.userData = featureData.objInfos[objInfosIndex - 1];
-        } else {
+        if (featureData.starts.length === 1) {
             pickResult.userData = featureData.objInfos[0];
+            return;
+        }
+
+        const intersectIndex =
+            intersect.faceIndex !== undefined ? intersect.faceIndex * 3 : intersect.index!;
+
+        // TODO: Implement binary search.
+        let objInfosIndex = 0;
+        for (const featureStartIndex of featureData.starts) {
+            if (featureStartIndex > intersectIndex) {
```

Comment:
> This requires the feature data (TileFeatureData.starts array) to be sorted in ascending order is requirement always fulfilled?
Our decoders fulfill this requirement yes. In general these feature starts must be ordered otherwise they're useless. Every feature has multiple faces/vertices. Feature i starts at starts[i] and ends at starts[i+1]-1. If they weren't ordered you wouldn't know where a feature ends.
Anyway, I'll add a comment in Geometry.featureStarts and TileFeatureData.starts to make this clear.

> Anyway I see a lot of danger in the search function: try to find next face.. pretend that previous was the exact choice...
As I said, that's exactly how feature starts are arranged, the beginning of one feature marks the end of the previous one.

> Shouldn't it be:
> 
> ```
> const objInfoIndex = featureData.starts!.findIndex(faceVertexIdx => {
>             return faceVertexIdx === intersect.faceIndex! * 3;
>         });
> ```
No, that doesn't work, the vertex you're searching for might not be the start of any feature, but be somewhere in the middle.